src/lib.rs

```diff
@@ -818,11 +818,18 @@ pub fn max_level() -> LevelFilter {
     unsafe { mem::transmute(MAX_LOG_LEVEL_FILTER.load(Ordering::Relaxed)) }
 }
 
+#[cfg(feature = "use_std")]
+pub fn try_set_logger<M>(make_logger: M) -> Result<(), SetLoggerError>
+    where M: FnOnce(MaxLevelFilter) -> Box<Log>
+{
+    unsafe { set_logger_raw(|max_level| mem::transmute(make_logger(max_level))) }
+}
+
 /// Sets the global logger.
 ///
 /// The `make_logger` closure is passed a `MaxLevel` object, which the
 /// logger should use to keep the global maximum log level in sync with the
-/// highest log level that the logger will not ignore.
+/// highest log level that the logger wills not ignore.
 ///
 /// This function may only be called once in the lifetime of a program. Any log
```

Comment:
You should mention that further calls to `set_logger` after the first will panic.